internal/fwschemadata/data_set_test.go

```diff
@@ -139,6 +139,39 @@ func TestDataSet(t *testing.T) {
 				),
 			}),
 		},
+		"write-tftypes-values": {
```

Comment:
Same as above, we should add a test for the error scenario of the `tftypes.Value` being the wrong type (like an object missing one of the attributes)